static/src/javascripts/projects/common/modules/commercial/acquisitions-link-enrichment.js

```diff
@@ -51,6 +51,27 @@ const addReferrerDataToAcquisitionLinksOnPage = (): void => {
     });
 };
 
+const addReferrerDataToAcquisitionLinksInIframes = (): void => {
+    window.addEventListener('message', event => {
+        let data;
+        try {
+            data = JSON.parse(event.data);
+        } catch (e) {
+            return;
+        }
+
+        // Expects enrich requests to be made via iframe-messenger:
+        // https://github.com/guardian/iframe-messenger
+        if (data.type === 'enrich-acquisition-links' && data.id) {
+            data.referrerData = addReferrerData({});
+            [...document.getElementsByTagName('iframe')].forEach(iframe => {
+                iframe.contentWindow.postMessage(data);
```

Comment:
I think you're missing a second argument here, the `targetOrigin`:

https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage

This could be used to limit the number of messages we send out and make sure third-parties don't get receive these messages